fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PruneAggChildColumns.java

```diff
@@ -54,15 +58,57 @@ public class PruneAggChildColumns extends OneRewriteRuleFactory {
     @Override
     public Rule build() {
         return RuleType.COLUMN_PRUNE_AGGREGATION_CHILD.build(logicalAggregate().then(agg -> {
+            Slot slot = handleCountStar(agg);
+            List<Slot> childOutput = agg.child().getOutput();
+            if (slot != null) {
+                if (childOutput.size() == 1 && childOutput.get(0).equals(slot)) {
+                    return agg;
+                }
+                return agg.withChildren(ImmutableList.of(new LogicalProject<>(ImmutableList.of(slot), agg.child())));
+            }
             List<Expression> slots = Lists.newArrayList();
             slots.addAll(agg.getExpressions());
             Set<Slot> outputs = SlotExtractor.extractSlot(slots);
-            List<NamedExpression> prunedOutputs = agg.child().getOutput().stream().filter(outputs::contains)
+            List<NamedExpression> prunedOutputs = childOutput.stream().filter(outputs::contains)
                     .collect(Collectors.toList());
             if (prunedOutputs.size() == agg.child().getOutput().size()) {
                 return agg;
             }
             return agg.withChildren(ImmutableList.of(new LogicalProject<>(prunedOutputs, agg.child())));
         }));
     }
+
+    private Slot handleCountStar(LogicalAggregate<GroupPlan> agg) {
+        List<NamedExpression> outputExpressions = agg.getOutputExpressions();
+        if (outputExpressions.size() != 1) {
+            return null;
+        }
+        NamedExpression namedExpression = outputExpressions.get(0);
```

Comment:
we need to collect all agg function from all output expressions and judge whether all of them have no child output slot reference as leaf node